ml-algorithms/src/main/java/org/opensearch/ml/engine/indices/MLIndicesHandler.java

```diff
@@ -52,6 +56,21 @@ public class MLIndicesHandler {
         }
     }
 
+    /**
+     * Determines whether an index exists on non-multi tenancy enabled environments. Otherwise,
+     * returns true when multiTenancy is Enabled
+     *
+     * @param clusterService the cluster service
+     * @param isMultiTenancyEnabled whether multi-tenancy is enabled
+     * @param indexName - the index to search
+     * @return boolean indicating the existence of an index. Returns true if multitenancy is enabled.
+     * @implNote It is recommended that if your environment enables multi tenancy, your plugin indices are
+     * pre-populated otherwise this method will result in unwanted early returns without checking the clusterService
```

Comment:
Wasn't aware of this javadoc.  I like it... more than "recommended" I'd say it's _assumed_ that when multitenancy is enabled that indices (or tables in DDB) are already created.